common/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php

```diff
@@ -75,6 +75,17 @@ public function validate(\Nethgui\Controller\ValidationReportInterface $report)
         ) {
             $report->addValidationErrorMessage($this, 'domain', 'valid_relay_notprimarydomain');
         }
+
+        // Domain validation, it must not be named like the first domain (FQDN)
+        $domainFQDN = $this->getPlatform()->getDatabase('configuration')->getType('DomainName');
+        $domain = $this->parameters['domain'];
+
+        if($this->getRequest()->isMutation()
+            && preg_match ("/$domainFQDN/", $domain)
```

Comment:
Does `preg_match` match substrings too? I'd implement a case insensitive, full-string comparision (`strtolower` both terms).

Reply:
yes true...I need to compare with a lowercase.
> Does preg_match match substrings too?

what do you means ?

if the FQDN is domain.com we want to prevent the sub.domain.com is used, so yes it is forbidden